components/StepsProgress.js

```diff
@@ -107,43 +207,71 @@ const StepsProgress = ({
   onStepSelect,
   allCompleted,
   stepWidth,
+  forceMobile,
 }) => {
   const focusIdx = focus ? steps.findIndex(step => step.name === focus.name) : -1;
+  const mobileStepIdx = allCompleted ? steps.length - 1 : focusIdx > -1 ? focusIdx : 0;
+  const mobileNextStepText = mobileStepIdx < steps.length - 1 ? steps[mobileStepIdx + 1].name : 'A small gift for you!';
+
+  // TODO: move to pie progress to a separate comp?
+  const progress = allCompleted ? 100 : (100 / steps.length) * (mobileStepIdx + 1);
+  const color = '#1F87FF';
+  const bgColor = '#D9DBDD';
+  const pieSize = '48';
 
   return (
-    <Flex className="steps-progress">
-      {steps.map((step, idx) => {
-        const stepName = step.name;
-        const checked = idx < focusIdx || allCompleted;
-        const focused = idx === focusIdx;
-        const disabled = disabledStepNames.includes(stepName);
-        const loading = loadingStep && stepName === loadingStep.name;
-
-        return (
-          <Flex
-            key={stepName}
-            className={classNames(`step-${stepName}`, { disabled })}
-            flexDirection="column"
-            alignItems="center"
-            css={{ flexGrow: 1, flexBasis: stepWidth }}
-          >
-            <Flex alignItems="center" mb={2} css={{ width: '100%' }}>
-              <SeparatorLine active={checked || focused} transparent={idx === 0} />
-              <Bubble
-                disabled={disabled}
-                onClick={disabled ? undefined : onStepSelect && (() => onStepSelect(step))}
-                checked={checked}
-                focus={focused}
-              >
-                {getBubbleContent(idx, checked, loading)}
-              </Bubble>
-              <SeparatorLine active={checked} transparent={idx === steps.length - 1} />
+    <StepsOuter className="steps-progress">
+      {forceMobile ? (
+        <StepMobile>
+          <StepsMobileLeft>
+            <p>{steps[mobileStepIdx].name}</p>
+            <p>Next: {mobileNextStepText}</p>
+          </StepsMobileLeft>
+          <StepsMobileRight>
+            <PieProgressWrapper>
+              <PieProgress progress={progress} pieSize={pieSize}>
+                <PieHalfCircleLeft progress={progress} pieSize={pieSize} />
+                <PieHalfCircleRight progress={progress} pieSize={pieSize} />
+              </PieProgress>
+              <PieShadow pieSize={pieSize} bgColor={bgColor} />
+            </PieProgressWrapper>
+            {mobileStepIdx + 1} of {steps.length}
```

Comment:
Needs to be internationalized. See https://docs.opencollective.com/help/developers/translations